diff --git a/dist/shiki.js b/dist/shiki.js
index 49fe908d2cc83b9a63d504c2d9a468de5a6fa3b3..832fae4bcea8be5d28be405c26f2b443f12e8f19 100644
--- a/dist/shiki.js
+++ b/dist/shiki.js
@@ -30,6 +30,8 @@ async function createShikiHighlighter({
   return {
     async highlight(code, lang = "plaintext", options) {
       const loadedLanguages = highlighter.getLoadedLanguages();
+      const isTerminal = lang === "terminal";
+      if (isTerminal) lang = "plaintext";
       if (!isSpecialLang(lang) && !loadedLanguages.includes(lang)) {
         try {
           await highlighter.loadLanguage(lang);
@@ -65,6 +67,7 @@ async function createShikiHighlighter({
               const classValue = (normalizePropAsString(node.properties.class) ?? "") + (attributesClass ? ` ${attributesClass}` : "");
               const styleValue = (normalizePropAsString(node.properties.style) ?? "") + (attributesStyle ? `; ${attributesStyle}` : "");
               node.properties.class = classValue.replace(/shiki/g, "astro-code");
+              if (isTerminal) node.properties.class += " lang-terminal";
               node.properties.dataLanguage = lang;
               if (wrap === false) {
                 node.properties.style = styleValue + "; overflow-x: auto;";
