diff --git a/dist/shiki.js b/dist/shiki.js
index 4600b82f60d94b32692f3e14a298e237df77540d..fed90d96bb85966c2e26086d86788780250f5d3a 100644
--- a/dist/shiki.js
+++ b/dist/shiki.js
@@ -20,6 +20,8 @@ async function createShikiHighlighter({
     themes: Object.values(themes).length ? Object.values(themes) : [theme]
   });
   async function highlight(code, lang = "plaintext", options, to) {
+    const is_terminal = lang === "terminal";
+    if (is_terminal) lang = "plaintext";
     const resolvedLang = langAlias[lang] ?? lang;
     const loadedLanguages = highlighter.getLoadedLanguages();
     if (!isSpecialLang(lang) && !loadedLanguages.includes(resolvedLang)) {
@@ -57,6 +59,7 @@ async function createShikiHighlighter({
             const classValue = (normalizePropAsString(node.properties.class) ?? "") + (attributesClass ? ` ${attributesClass}` : "");
             const styleValue = (normalizePropAsString(node.properties.style) ?? "") + (attributesStyle ? `; ${attributesStyle}` : "");
             node.properties.class = classValue.replace(/shiki/g, "astro-code");
+            if (is_terminal) node.properties.class += " lang-terminal";
             node.properties.dataLanguage = lang;
             if (options.wrap === false || options.wrap === void 0) {
               node.properties.style = styleValue + "; overflow-x: auto;";
